pinot-core/src/test/java/org/apache/pinot/queries/AllNullQueriesTest.java

```diff
@@ -660,4 +661,12 @@ public void testQueries(ColumnDataType columnDataType, File indexDir, Query quer
     _indexSegment.destroy();
     FileUtils.deleteDirectory(indexDir);
   }
+
+  @AfterMethod
+  public void destroySegments() {
+    _indexSegment.destroy();
+    for (IndexSegment indexSegment : _indexSegments) {
```

Comment:
(minor) Closing `_indexSegment` itself should be enough because `_indexSegments` is simply references. Same for other places